service/qos/service_level_controller.cc

```diff
@@ -326,26 +326,29 @@ future<> service_level_controller::alter_distributed_service_level(sstring name,
 }
 
 future<> service_level_controller::drop_distributed_service_level(sstring name, bool if_exists) {
-   auto sl_info = co_await _sl_data_accessor->get_service_levels();
-        auto it = sl_info.find(name);
-        if (it == sl_info.end()) {
-            if (if_exists) {
-                co_return;
-            } else {
-                throw nonexistant_service_level_exception(name);
-            }
-        } 
-            auto& role_manager = _auth_service.local().underlying_role_manager();
-            auto attributes = co_await role_manager.query_attribute_for_all("service_level");
-                co_await coroutine::parallel_for_each(attributes.begin(), attributes.end(), [&role_manager, name] (auto&& attr) {
-                    if (attr.second == name) {
-                        return do_with(attr.first, [&role_manager] (const sstring& role_name) {return role_manager.remove_attribute(role_name, "service_level");});
-                    } else {
-                        return make_ready_future();
-                    }
-                });
-            
-                co_return co_await _sl_data_accessor->drop_service_level(name);
+    auto sl_info = co_await _sl_data_accessor->get_service_levels();
+    auto it = sl_info.find(name);
+    if (it == sl_info.end()) {
+        if (if_exists) {
+            co_return;
+        } else {
+            throw nonexistant_service_level_exception(name);
+        }
+    }
+    
+    auto& role_manager = _auth_service.local().underlying_role_manager();
+    auto attributes = co_await role_manager.query_attribute_for_all("service_level");
+    co_await coroutine::parallel_for_each(attributes.begin(), attributes.end(), [&role_manager, name] (auto&& attr) {
+        if (attr.second == name) {
+            return do_with(attr.first, [&role_manager] (const sstring& role_name) {
+                return role_manager.remove_attribute(role_name, "service_level");
+            });
```

Comment:
Nit: `do_with` is used to prolong lifetime of objects until the future returned by the inner lambda completes. While this is useful for continuation style code, with coroutines it is pretty useless, so it could have been removed.